curriculum/challenges/english/25-front-end-development/quiz-javascript-functional-programming/66edd4f31ff19bf5573bf64b.md

```diff
@@ -17,439 +17,472 @@ Answer all of the questions below correctly to pass the quiz.
 
-Placeholder distractor 3
+```javascript
+let count = 0;
+const increment = () => ++count;
+```
 
 #### --answer--
 
-Placeholder answer
+```javascript
+const double = (x) => x * 2;
+```
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the primary goal of currying?
 
 #### --distractors--
 
-Placeholder distractor 1
+To improve code readability
 
 ---
 
-Placeholder distractor 2
+To create functions with fewer arguments
 
 ---
 
-Placeholder distractor 3
+To eliminate the need for higher-order functions
 
 #### --answer--
 
-Placeholder answer
+To enable partial application of functions
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which statement about side effects is FALSE?
 
 #### --distractors--
 
-Placeholder distractor 1
+Side effects make code harder to test
 
 ---
 
-Placeholder distractor 2
+Modifying a global variable is a side effect
 
 ---
 
-Placeholder distractor 3
+Reading from a database can be considered a side effect
 
 #### --answer--
 
-Placeholder answer
+Pure functions can have side effects
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the result of currying the following function?
+
+```javascript
+const add = (a, b, c) => a + b + c;
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+```javascript
+const curried = (a, b, c) => a() + b() + c();
+```
 
 ---
 
-Placeholder distractor 2
+```javascript
+const curried = a => (b, c) => a + b + c;
+```
 
 ---
 
-Placeholder distractor 3
+```javascript
+const curried = (a, b) => c => a + b + c;
+```
 
 #### --answer--
 
-Placeholder answer
+```javascript
+const curried = a => b => c => a + b + c;
+```
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is NOT a side effect?
 
 #### --distractors--
 
-Placeholder distractor 1
+Modifying the DOM
 
 ---
 
-Placeholder distractor 2
+Sending an HTTP request
 
 ---
 
-Placeholder distractor 3
+Changing the value of a global variable
 
 #### --answer--
 
-Placeholder answer
+Creating a new object and returning it
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is a higher-order function?
 
 #### --distractors--
 
-Placeholder distractor 1
+A function that takes longer to execute
 
 ---
 
-Placeholder distractor 2
+A function with more than three arguments
 
 ---
 
-Placeholder distractor 3
+A function that uses advanced JavaScript features
 
 #### --answer--
 
-Placeholder answer
+A function that returns another function
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following best describes a pure function?
 
 #### --distractors--
 
-Placeholder distractor 1
+A function that doesn't use any variables
 
 ---
 
-Placeholder distractor 2
+A function that only uses arrow function syntax
 
 ---
 
-Placeholder distractor 3
+A function that doesn't have any parameters
 
 #### --answer--
 
-Placeholder answer
+A function that always returns the same output for the same input and has no side effects
```

Comment:
This answer is very similar to answer for first question. Maybe it could be reworded? Or question could be replaced with another one?

Reply:
Hey, I have changed this question!